waku.md

```diff
@@ -64,7 +64,17 @@ Waku was created to incrementally improve in areas that Whisper is lacking in, w
 | **Mail Server** | A node responsible for archiving messages.       |
 | **Mail Client** | A node that requests messages from mail servers. |
 
-## Specification
+## Underlying Transports and Prerequisites
+
+### Use of DevP2P
+
+For nodes to communicate, they MUST implement devp2p and run RLPx. They MUST have some way of connecting to other nodes. Node discovery is largely out of scope for this spec, but see the appendix for some suggestions on how to do this.
+
+### Gossip based routing
+
+In Whisper, messages are gossiped between peers. They float around the network until their TTL has expired. A node SHOULD relay messages to all connected nodes if an envelope matches their PoW and bloom filter settings. If a node works in light mode, it MAY choose not to forward envelopes. A node MUST NOT send expired envelopes, unless the envelopes are sent as a mailserver response.
```

Comment:
if they should be relayed to all connected nodes then its not really gossiping is it?

Reply:
Good point, rather flooding I guess.

Reply:
Flooding is also a form of gossiping, isn't it? It's also based on matching bloom filters. Added clarification here though re flooding and rumor mongering.